README.md

```diff
@@ -29,35 +43,165 @@ Then add this snippet to your `build.sbt`
+`Deduplication` instance.
+
+Once you have an instance of `DeduplicationContext`, you can use
+`context.protect(id: ID, fa: F[A])` to wrap your side effects.
+
+#### DynamoDB backend
+
+In order to create a new `Deduplication` instance you will need an
+implementation of `ProcessRepo`, which is the data backend that is responsible
+to store information around which operations have been executed and what their
+results were.
+
+The library has a built-in implementation of `ProcessRepo` that uses DynamoDB
+and [Meteor](https://d2a4u.github.io/meteor/) under the hood. You can use it by
+creating an instance of `Deduplication` with the
+`com.ovoenergy.comms.deduplication.meteor.MeteorDeduplication` factory object.
+
+If you use the meteor implementation you will need to make sure the meteor
+codecs for the return type of your operations are in scope when you instantiate
+new contexts.
+
+#### Example
+
+In the following example a service is consuming a stream of events and
+performing 2 effectful operations on top of them.
 
-A `processorId` needs to be assigned to each instance of this library. It will uniquely identify the processor. If two services have the same `processorId` it likely means they are two instances of the same service.
+Using the library we make sure[*](#limitations) that, if the same event is
+consumed more than once, the side effects will not be re-executed.
 
-We need to know the max amount of time the process will take (`maxProcessingTime` in the config). Any process that take more than this amount of time will be considered dead.
+```scala
+import cats.effect._
+import com.ovoenergy.comms.deduplication
+import com.ovoenergy.comms.deduplication.meteor.MeteorDeduplication
+import com.ovoenergy.comms.deduplication.meteor.codecs._
+import meteor.CompositeKeysTable
+import meteor.syntax._
+
+// A stream of events.
+// This could contain duplicates itself or the same event could be present
+// in different streams across service instances
+val events: Stream[IO, MyEvent] = ???
+
+// Effectful operations that need to be executed for each event
+def sendEmail(evt: MyEvent): IO[String] = ???
+def storeEmail(evt: MyEvent, sendId: String): IO[Unit] = ???
+
+// Global configuration
+val dedupConf: deduplication.Config = ???
+val dedupTable: CompositeKeysTable[String, String] = ???
+
+val dedupResource = for {
+  client <- meteor.Client.resource[IO]
+  dedup <- MeteorDeduplication.resource[IO, String, String](
+    client,
+    dedupTable,
+    dedupConf
+  )
+} yield dedup
+
+dedupResource.use { deduplication =>
+
+  // Create two contexts to deduplicate the operations separately
+  val sendEmailCtx = deduplication.context[String]("sendEmail")
+  val storeEmailCtx = deduplication.context[Unit]("storeEmail")
+
+  events
+    .evalMap { evt =>
+      for {
+        // Wrap the operations in a protect call
+        sendId <- sendEmailCtx.protect(evt.id, sendEmail(evt))
```

Comment:
I think that if I just jumped into this example I could have the following question:
> what happens when `sendEmailCtx.protect(evt.id, sendEmail(evt)` detects a duplicate? Does it fail the `F` thus leading to short-circuiting behaviour?

Reply:
Good shout 👍